test/sql/parser/digit_separators.test

```diff
@@ -0,0 +1,175 @@
+# name: test/sql/parser/digit_separators.test
+# group: [parser]
+
+query I
+SELECT 1__2
```

Comment:
Python only allows individual underscores, not multiple underscores following each other, e.g. this would throw an error in Python:

```py
>>> 1__2
  File "<stdin>", line 1
    1__2
     ^
SyntaxError: invalid decimal literal
```

I'm not too opposed to keeping support for multiple subsequent underscores but it does allow for constructing very unreadable numbers.